addons/website_event_crm/models/event_lead_rule.py

```diff
@@ -0,0 +1,25 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import models
+from odoo.http import request
+
+
+class EventLeadRule(models.Model):
+
+    _inherit = "event.lead.rule"
+
+    def _run_on_registrations(self, registrations):
+        leads = super()._run_on_registrations(registrations)
+        if len(self.env['res.lang'].get_installed()) < 2:
+            return leads
+
+        for lead in leads:
+            # When a lead has several registrations, they should all relate to the same visitor
+            visitors = lead.registration_ids.visitor_id
+            if visitors:
+                lead.visitor_ids |= visitors
+            if not lead.lang_id and request and request.website.language_count > 1:
+                lead.lang_id = request.lang.id
```

Comment:
@adrienmilis 

About the language: it makes sense since it is the same use of request.lang in _upsert_vistor(). It seems efficient :)

About the condition: Could there be no request? I am also not sure about checking the language_count either.

Cheers :)